### ISSUE TITLE
BSD/OS 4.0.1 needs -DSTRUCT_TM_HASZONE for POSIX.so

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#82]
Migrated from [rt.perl.org#885](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=885) (status was 'resolved')

Searchable as RT885$

**Comment by p5pRT** [comment]
### From mab@alink.net



BSD/OS has had tm\_zone in struct tm for I don't know how long\, but as
of exactly BSD/OS 4\.0\.1 you \*must\* compile POSIX\.so with
\-DSTRUCT\_TM\_HASZONE or else strftime's %Z will output garbage and/or crash\.

hints/bsdos\.sh needs this&#8203;:

\*\*\* bsdos\.sh    1999/01/28 07&#8203;:45&#8203;:32     1\.1\.1\.1\.2\.1
\-\-\- bsdos\.sh    1999/06/19 04&#8203;:38&#8203;:29     1\.1\.1\.1\.2\.2
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 89\,94 \*\*\*\*
\-\-\- 89\,95 \-\-\-\-
&nbsp;        libswanted="rpc curses termcap $libswanted"
&nbsp;        ;;
&nbsp;  4\.0\*\)
\+       POSIX\_cflags='ccflags="$ccflags \-DSTRUCT\_TM\_HASZONE"' 
&nbsp;        \# ELF dynamic link libraries starting in 3\.1
&nbsp;          useshrplib='true'
&nbsp;        so='so'

<details><summary>Perl Info</summary>

```


Site configuration information for perl 5.00404:

Configured by mab at Wed Jan 27 23:04:01 PST 1999.

Summary of my perl5 (5.0 patchlevel 4 subversion 4) configuration:
  Platform:
    osname=bsdos, osvers=4.0, archname=i386-bsdos
    uname='bsdos hh.alink.net 4.0 bsdi bsdos 4.0 kernel #0: thu jan 21 15:01:06 pst 1999 mab@hh.alink.net:usrsrcsyscompilealink i386 '
    hint=previous, useposix=true, d_sigaction=define
    bincompat3=y useperlio=undef d_sfio=undef
  Compiler:
    cc='cc -g', optimize='-O2', gccversion=2.7.2.1
    cppflags='-I/usr/local/include'
    ccflags ='-I/usr/local/include'
    stdchar='char', d_stdstdio=undef, usevfork=true
    voidflags=15, castflags=0, d_casti32=define, d_castneg=define
    intsize=4, alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='ld', ldflags =' -L/usr/X11/lib -L/usr/local/lib'
    libpth=/usr/local/lib /usr/shlib /shlib /usr/lib /usr/X11/lib
    libs=-ldl -lm -lc
    libc=/shlib/libc.so, so=so
    useshrplib=true, libperl=libperl.so
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic -Wl,-rpath,/usr/libdata/perl5/i386-bsdos/5.00404/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -x  -L/usr/X11/lib -L/usr/local/lib'

Locally applied patches:
	


@INC for perl 5.00404:
	/usr/libdata/perl5/i386-bsdos/5.00404
	/usr/libdata/perl5
	/usr/libdata/perl5/site_perl/i386-bsdos
	/usr/libdata/perl5/site_perl
	.


Environment for perl 5.00404:
    HOME=/home/m/a/mab
    LANG=C
    LC_CTYPE=iso_8859_1
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/bin:/usr/local/krb5/bin:/bin:/usr/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/contrib/bin:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

fixed in change @&#8203;951

**Comment by p5pRT** [comment]
### From The RT System itself

fixed in change @&#8203;951
